code_docs/Organization.md

```diff
@@ -67,7 +75,13 @@ The documentation structure would look something like this:
   - Rename `PerformanceSessionView` to `PerformanceView`
 - `AutomationView` is too large to manage
 - **Proposal:**
-  - Refactor `AutomationView` to a base-class which can be inherited from in different contexts, behavior can be organized in child-classes
+  - Create `AutomationEditor` which is just the editor itself. This class can be set up with different layouts (unipolar/bipolar/note velocity/etc) depending on the parameter being edited
+  - Create `AutomationEditorLaunchpad` which holds the current Automation Overview. This has three context-based varieties: arranger automation, clip automation note-row automation
 
 #### UI
-- There's a whole bunch of global stuff here. It would probably help legibility in other parts of the code if these were captured in a (static) class. UI mode defines could be in an enum : uint32_t for the same reason.
+- `UI.h` There's a whole bunch of global stuff here. It would probably help legibility in other parts of the code if these were captured in a (static) class. UI mode defines could be in an enum : uint32_t for the same reason.
+- In several files in `ui/` there are long lists of includes, namespaces and class calls. Each of these should be mapped out to find out what exactly is happening, and how we can simplify this
+##### RootUI
+It is unclear what the exact purpose of `RootUI` is. It basically just implements a list of virtual functions. The only way `RootUI` is used, is as the base-class of `TimelineView`, which we discussed above should be reworked into base-class `View`. This means these virtual function declarations should probably just move to `View` so we can lose `RootUI` altogether.
+
+`getRootUI()` is declared in a bunch of places, which should probably be refactored to `getView()`, but more investigation into the entire UI management is needed before committing to any changes. `ui.cpp` does hold a few methods dealing with `rootUI`.
```

Comment:
rootUI is not just used with Views.

If you look at the SoundEditor UI, which is the main class that handles the menu system, the root UI is used to check what the underlying view. The current UI while in the menu would be the sound editor UI.

This exists because while in the menu you can still interact with the underlying UI in certain situations. E.g. using the audition pads, or the keyboard (in keyboard view) while in a parameter menu.

You'll see a lot of checks to see what the root UI is while you're in the menu.

Reply:
Yeah I probably could have worded this a bit more precisely, but currently I'm kind of writing it for my own understanding with the intention of ultimately refining it for "public consumption"

So what I'm trying to get around here is, what is the function of RootUI existing in the context of the entire codebase? Do we need RootUI to exist or can it be absorbed into other classes for simplification? Especially because it adds so little in terms of implementation.

When I say RootUI is only used with views, what I mean to say is: views are the only classes that inherit from RootUI. And having already discussed that views can use a good refactor anyway, we could easily absorb RootUI into the new View class without really touching any of the other code.

At line 87 I do mention the getRootUI() method, which would need to get refactored into View.

Reply:
I think the context might be the following:

The UI system in the Deluge actually has the capability of having multiple levels.

That is to say that you could start with a base (root UI), and then enter another UI, and then enter another UI. 

So the root UI always points to the lowest level UI that if you back out all the way of the current UI it will bring you to the root UI.

To your point, I think a view will always be a root UI as you would not "back out" of one view to go to another view. You can however swap a root UI at level (as is done in the menu when accessing automation view when instrument clip view is the root UI).

Ultimately I think view would inherit from UI.

The UI class keeps track of the hierarchy of UI's that have been opened.

To me getRootUI, changeRootUI belongs in the UI class.

Reply:
> To your point, I think a view will always be a root UI as you would not "back out" of one view to go to another view. You can however swap a root UI at level (as is done in the menu when accessing automation view when instrument clip view is the root UI).

Agreed

> To me getRootUI, changeRootUI belongs in the UI class.

Yes, makes a lot more sense